runtime/src/beefy.rs

```diff
@@ -0,0 +1,194 @@
+	where
+		BeefyAuthoritySet<H>: parity_scale_codec::Decode,
+	{
+		/// Gets the current beef stakes
+		fn current_beef_stakes() -> BeefStakes<Runtime>;
+
+		/// Gets the next beef stakes
+		fn next_beef_stakes() -> BeefStakes<Runtime>;
+
+		/// Returns the authority set based on the current beef stakes
+		fn compute_current_authority_set(
+			beef_stakes: BeefStakes<Runtime>,
+		) ->  BeefyAuthoritySet<H>;
+
+		/// Returns the authority set based on the next beef stakes
+		fn compute_next_authority_set(
+			beef_stakes: BeefStakes<Runtime>,
+		) -> BeefyNextAuthoritySet<H>;
+	}
+}
+
+pub fn current_beef_stakes() -> BeefStakes<Runtime> {
+	// Similar set of validators of pallet beefy fn validator_set();
+	// the benefit of this is being an unwrapped value of Vec<Public>
+	let current_validators = pallet_beefy::pallet::Authorities::<Runtime>::get().to_vec();
+
+	let current_committee = SessionValidatorMngPallet::<Runtime>::current_committee_storage();
+
+	compute_beef_stakes(current_validators, current_committee)
+}
+
+pub fn next_beef_stakes() -> BeefStakes<Runtime> {
+	let next_validators = pallet_beefy::pallet::NextAuthorities::<Runtime>::get().to_vec();
+
+	match SessionValidatorMngPallet::<Runtime>::next_committee_storage() {
+		Some(next_committee) => {
+			log::info!("XXXXXXXXXXXXXXXXXXXXXXXXXXXX NEXT VALIDATORS: {next_validators:?}");
+
+			compute_beef_stakes(next_validators, next_committee)
+		},
+		None => {
+			log::info!("XXXXXXXXXXXXXXXXXXXXXXXXXXXX No Next Committee, setting stakes to 0");
+			next_validators.into_iter().map(|v| (v, 0)).collect()
+		},
+	}
+}
+
+pub fn compute_current_authority_set(
+	beef_stakes: BeefStakes<Runtime>,
+) -> BeefyAuthoritySetOf<Runtime> {
+	// get the validator set id
+	let authority_proof = BeefyMmrPallet::<Runtime>::authority_set_proof();
+	let id = authority_proof.id;
+
+	compute_authority_set(id, beef_stakes)
+}
+
+pub fn compute_next_authority_set(
+	beef_stakes: BeefStakes<Runtime>,
+) -> BeefyAuthoritySetOf<Runtime> {
+	let authority_proof = BeefyMmrPallet::<Runtime>::next_authority_set_proof();
+	let id = authority_proof.id;
+
+	compute_authority_set(id, beef_stakes)
+}
+
+fn compute_beef_stakes(
+	validators: Vec<BeefyIdOf<Runtime>>,
+	committee: CommitteeInfoOf<Runtime>,
+) -> BeefStakes<Runtime> {
+	let mut committee_members = committee.committee;
+
+	let mut beefy_with_stakes = Vec::new();
+
+	for validator in validators {
+		log::info!("XXXXXXXXXXXXXXXXXXXXXXXXXXXX Check if Validator({validator}): in committee:");
+		let position = committee_members.iter().position(|elem| match elem {
+			CommitteeMember::Permissioned { id, .. } => {
+				// convert to beefy
+				let committee_id = xchain_public_to_beefy(id.clone());
+				if committee_id == validator {
+					beefy_with_stakes.push((
+						committee_id,
+						// default stake
+						1,
+					));
+					// remove from the list
+					true
+				} else {
+					false
+				}
+			},
+			CommitteeMember::Registered { .. } => false,
```

Comment:
I'm thinking we should try to retrieve these stakes now, or comment + log error so we have more of a reminder to come back soon to get the stakes.